code/game/objects/structures/barricade.dm

```diff
@@ -482,7 +482,7 @@
 
 		balloon_alert_to_viewers("repairing base...")
 
-		if(!do_after(user, 2 SECONDS, TRUE, src, BUSY_ICON_FRIENDLY) || obj_integrity >= max_integrity)
+		if(!do_after(user, 2 SECONDS, TRUE, src, BUSY_ICON_FRIENDLY) || obj_integrity >= max_integrity * 0.3)
```

Comment:
I don't see how it allowed instant repairing before, since failing the doafter would return out regardless. Are you talking about stacking doafters to repair faster? If you are, why not check do_actions instead of this?

Reply:
I mean both solutions work

Reply:
Also yes it's done by spam clicking the cade, stacking do_afters

Reply:
> Also yes it's done by spam clicking the cade, stacking do_afters

You can make things only able to have one do-after by giving them a unique interaction key in the do_after arguments

Reply:
Yeah welder repairs are already limited to a single drafter, metal/plasteel repair should be consistent.

Reply:
We don't have that in tgmc code

Reply:
> We don't have those interaction keys* in tgmc code

very sad.